### PR TITLE
feat: governance on code

### PR DESCRIPTION
**Pull request by manuelurgell**
Added pre-commit and some hooks to ensure governance on code.

dvc-repro runs successfully:
Screenshot 2024-11-02 at 1 50 43 p m

mlflow:
Screenshot 2024-11-02 at 1 52 33 p m

### CODE REVIEW

File: pipeline/model/classifiermodel.py [hunk 1]
```diff
@@ -72,98 +84,131 @@ def cleanAndPreprocessData(self, X: DataFrame, y: DataFrame):
...
 
-        # Normalizar datos 
-        normalizer = MinMaxScaler()
-        # Aplicar la normalización 
-        XT_normalized = pd.DataFrame(normalizer.fit_transform(XT), columns=XT.columns)
+        # Normalizar datos
+        # normalizer = MinMaxScaler()
+
+        # Aplicar la normalización
+        # XT_normalized = pd.DataFrame(
```
Comment (Immutare) [inline pipeline/model/classifiermodel.py]: Por que comentaste esto si no estaba comentado?

File: pipeline/model/classifiermodel.py [hunk 2]
```diff
@@ -72,98 +84,131 @@ def cleanAndPreprocessData(self, X: DataFrame, y: DataFrame):
...
+    def splitTrainTestVal(
+        self,
+        X: DataFrame,
+        y: DataFrame,
+        trainSize: float = 0.8,
+        testSize: float = 0.15,
+    ):
         dataSize = trainSize + testSize
-        if (dataSize > 1):
+        if dataSize > 1:
```
Comment (Immutare) [inline pipeline/model/classifiermodel.py:130]: Esto esta bien? Es parte del estandar de codigo?

**Review by Immutare — CHANGES_REQUESTED** [review]
Everything seems to be ok but found a couple of lines that shouldn't be commented, please review.

File: pipeline/model/classifiermodel.py (hunk 1 above)
Comment (manuelurgell) [reply pipeline/model/classifiermodel.py]: Son variables no usadas

File: pipeline/model/classifiermodel.py (hunk 2 above)
Comment (manuelurgell) [reply pipeline/model/classifiermodel.py:130]: Si, en python las condiciones no llevan paréntesis a menos que cubran varias líneas

**Review by Immutare — APPROVED** [review]
Approved, thanks!

**Pull request merged**